pkg/aws/awscontrolplane/validate_awscontrolplane.go

```diff
@@ -53,81 +53,65 @@ func NewValidator(config config.Config) (*Validator, error) {
 
 func (v *Validator) Validate(request *admissionv1.AdmissionRequest) (bool, error) {
 	var awsControlPlane infrastructurev1alpha2.AWSControlPlane
+	var g8sControlPlane *infrastructurev1alpha2.G8sControlPlane
 	var err error
 
 	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, &awsControlPlane); err != nil {
 		return false, microerror.Maskf(parsingFailedError, "unable to parse awscontrol plane: %v", err)
 	}
-	err = v.AZValid(awsControlPlane)
-	if err != nil {
-		return false, microerror.Mask(err)
-	}
 	err = v.AZCount(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
-	err = v.InstanceTypeValid(awsControlPlane)
+	err = v.AZValid(awsControlPlane)
 	if err != nil {
 		return false, microerror.Mask(err)
 	}
-	err = v.AZOrder(request)
-	if err != nil {
-		return false, microerror.Mask(err)
+	// The order can only change on update
+	if request.Operation == admissionv1.Update {
```

Comment:
The code complexity is kind of increasing with this. A better design would be to have a clear separation from the beginning so that the code paths are less spaghetti and more comprehensible. Let's talk about this if you like.

Reply:
Sure, we can talk about it any time :)
The check was formerly inside the validation function but for that we needed to pass the entire request.